### ISSUE TITLE
When making a new project, the center heading will show the wrong project title until user refreshes the page

### ISSUE THREAD

**Issue by jbaile20** [Paul-Austin-Oswego-CSC480-HCI521/OZ-CSC-480-HCI-521-Fall-2024#123]
## Problem
When the user creates a new project, if you select that project on the left panel _before_ refreshing the page, it will show the incorrect project title in the center of the screen. On a page refresh, it will show the correct title. 

### Steps to reproduce
- Create a new project on the left side panel
- Select the new project on the left panel
- The project title will be incorrect until a page refresh/revisit

### Environment
This happens when running locally _and_ on the moxie server
OS: Windows
Browser: Google Chrome

*Subheadings below are optional; remove if not applicable.*
### Location
any project URL

### Expected behavior
Project page shows the correct project title when clicked without needing to refresh
### Observed behavior
User needs to refresh the page before the correct project title is shown

*Subheadings below are optional; remove if not applicable.*
### Console output/errors
*Relevant console output or errors, pasted as text.*

### Screenshot
**before refresh, project 57 is selected but shows project 1**
![image](https://github.com/user-attachments/assets/f8c35016-93b0-480b-9894-909d05190fb3)
**after refresh, shows project 57 correctly**
![image](https://github.com/user-attachments/assets/20df575c-71f0-49db-86e7-a06131a6fe30)

*Top-level headings below are optional; remove if not applicable.*
## Suggested fix
N/A

## Other notes

*Remember to assign the issue exactly one severity label.*

**Comment by PiesArentSquare** [comment]
good catch! the issue is that project creation is handled in PageLayout, and so it does not update TaskPage's projects state. this would have to be fixed by either sending some message from PageLayout to TaskPage on project creation/edit/deletion forcing it to refetch, or probably a better solution is just using React.useContext at the top level
